### PR TITLE
featu: [Add support for exporting the configuration[ (FF-2437)

### PR DESCRIPTION
**Pull request by leoromanovsky**
[//]: #  (Link to the issue corresponding to this chunk of work)
Fixes: #__issue__

## Motivation and Context
[//]: #  (Why is this change required? What problem does it solve?)

Users who want to bootstrap their client SDK need a mechanism to export the configuration from their NodeJS server. 

feat store drawio

## Description
[//]: # (Describe your changes in detail)

Adds a new method to the client called `getConfiguration` - it is implemented by exporting the contents of the `servingStore: SyncStore`. This is implemented as a `Map` and trivial to export all the flags. The top-level method returns the UFC response.

Customers will be instructed to serialize this to JSON if desired and transmit it over an internal API.

## How has this been tested?
[//]: # (Please describe in detail how you tested your changes)

New unit tests.

[//]: # (OPTIONAL)
[//]: # (Add one or multiple labels: enhancement, refactoring, bugfix)

### CODE REVIEW

File: src/client/eppo-client.ts [hunk 1]
```diff
@@ -497,6 +499,14 @@ export default class EppoClient implements IEppoClient {
     this.isGracefulFailureMode = gracefulFailureMode;
   }
 
+  public getConfiguration(): IUniversalFlagConfig {
+    const flags = this.configurationStore.getAll();
+    const ufc: IUniversalFlagConfig = {
+      flags,
+    };
+    return ufc;
```
Comment (felipecsl) [inline src/client/eppo-client.ts:507]: you could simplify as:
```suggestion
    return { flags };
```

File: src/configuration-store/configuration-store.ts [hunk 2]
```diff
@@ -31,6 +32,7 @@ export interface IConfigurationStore<T> {
 
 export interface ISyncStore<T> {
   get(key: string): T | null;
+  getAll(): Record<string, T>;
```
Comment (felipecsl) [inline src/configuration-store/configuration-store.ts:35]: minor suggestion: a more idiomatic naming in typescript land would be `entries`, I think. Not a big deal though

**Review by felipecsl — APPROVED** [review]

**Review by sameerank — APPROVED** [review]

**Comment by sameerank** [comment]
Not sure if these questions are related, but here they go:

1. Isn't the Node server version of the configuration incompatible with the JS client version? Or if the diagram referring to using the JS client version on the server (which aligns with how I'm thinking about this) to get the obfuscated config?
2. And also we had paused exposing the ApiEndpoint class in [this PR](https://github.com/Eppo-exp/js-client-sdk-common/pull/73) but I think (in a separate PR) we can make it importable again. Does that sound right?

**Comment by leoromanovsky** [comment]
> Not sure if these questions are related, but here they go:
> 
> 
> 
> 1. Isn't the Node server version of the configuration incompatible with the JS client version? Or is the diagram referring to using the JS client version on the server (which aligns with how I'm thinking about this) to get the obfuscated config?

Yes this is a good observation. I am planning on allowing the JS client to configure itself in obfuscated or non obfuscated mode, the latter being for this use case of importing a configuration from nodejs. Do you see concerns with this approach, assuming the customer is aware of the security relaxation they are performing?

> 
> 2. And also we had paused exposing the ApiEndpoint class in [this PR](https://github.com/Eppo-exp/js-client-sdk-common/pull/73) but I think (in a separate PR) we can make it importable again. Does that sound right? Or does this approach make that unnecessary since the SDK is handling the fetch?

Cool yea now that we have the go ahead should complete that work to expose it.

File: src/configuration-store/configuration-store.ts [hunk 3]
```diff
@@ -41,6 +41,6 @@ export interface ISyncStore<T> {
 export interface IAsyncStore<T> {
   isInitialized(): boolean;
   isExpired(): Promise<boolean>;
-  getEntries(): Promise<Record<string, T>>;
```
Comment (leoromanovsky) [inline src/configuration-store/configuration-store.ts:42]: @felipecsl you had a review comment to rename `getAll` to `entries`. Ok for me to also rename `getEntries` to `entries` for consistency?

**Pull request merged**